tests/Cardano/SignerTests.cpp

```diff
@@ -0,0 +1,309 @@
+        hex(unisgnedEncodedCborData)
+    );
+
+    Data txId = Hash::blake2b(unisgnedEncodedCborData, 32);
+    EXPECT_EQ("d498c692e3101a39d19da9c7a7beccd65c7d1ea6d23008806ac8d46e81e4918f", hex(txId));
+
+    PrivateKey fromPriKey = PrivateKey(privateKey_b8c3);
+    PublicKey fromPubKey = fromPriKey.getPublicKey(TWPublicKeyTypeED25519Extended);
+    EXPECT_EQ(
+        "835610e371e632d829eb63bf44ec39d0487c19d4d98b8dce9dd88d88414ce5bb1ae60957699bf72b212ca823520cf7d86af5d1304cd90248fe60bd1fe442870f",
+        hex(fromPubKey.bytes)
+    );
+    Address fromAddr = Address(fromPubKey);
+    EXPECT_EQ("Ae2tdPwUPEZ6SqAETdiJgPYHpAey2MWakEVRDESWYzBePi7u5uAL5ah26qx", fromAddr.string());
+}
+
+TEST(CardanoSigner, SignTx_8283) {
+    // real transaction, with non-default fee
+    // see this tx: https://cardanoexplorer.com/tx/ebf58670ee1512d597876fe632e92790bd70568374bdbe5a69c5d8ed107607ee
+    Proto::SigningInput input;
+    input.set_amount((uint64_t)(8.0 * 1000000));
+    input.set_to_address("Ae2tdPwUPEYxKHuuYNsYkpR64XNZz7Xm8vMep7mZ2rrP9HgqbbeX1uMxmGd");
+    input.set_change_address("Ae2tdPwUPEYz548sTWdiTxBx13kxECHH4cmYtxQgPgEaQwmkymYFZzGkPrH");
+    auto utxo0 = input.add_utxo();
+    Data txid0 = parse_hex("a49bad3f69bbab0e4d3e51991ce7a1116c0fd322a7731246b92df455e67e6861");
+    utxo0->mutable_out_point()->set_txid(txid0.data(), txid0.size());
+    utxo0->mutable_out_point()->set_index(0);
+    utxo0->set_amount((uint64_t)(10.0 * 1000000));
+    input.add_private_key(privateKey_b8c3.data(), privateKey_b8c3.size());
+
+    Proto::TransactionPlan plan = Signer::planTransaction(input);
+    EXPECT_EQ("", plan.error());
+    // set non-default fee, 8000000 + 169884 + 1830116 = 10000000
+    EXPECT_EQ(167994, plan.fee());
+    EXPECT_EQ(1832006, plan.change());
+    EXPECT_EQ(10000000, plan.utxo(0).amount());
+    plan.set_fee((uint64_t)(0.169884 * 1000000));
+    plan.set_change(1830116);
+
+    Proto::SigningOutput output = Signer::sign(input, plan);
+
+    EXPECT_EQ(
+        "82839f8200d8185824825820a49bad3f69bbab0e4d3e51991ce7a1116c0fd322a7731246b92df455e67e686100ff9f8282d818582183581c0f1fdc075b3e562ff4c8dc6979b136aeedd087a7d3bf554f774b58eba0001abb7c9a121a007a12008282d818582183581c20b55a0bd5cbc39b58cdd8977e863aca7d387a5a0fc37a473eb72042a0001aa6fa2f8a1a001bece4ffa0818200d8185885825840835610e371e632d829eb63bf44ec39d0487c19d4d98b8dce9dd88d88414ce5bb1ae60957699bf72b212ca823520cf7d86af5d1304cd90248fe60bd1fe442870f584045b3fe46b52c839566e697cbe72e248cd0f01e10c7c5696c8f202fe9776602b63c5878f0e27869d1d9ff69c9773211406970742f7cc316690b9d9133794e4f0e",
+        hex(output.encoded())
+    );
+    EXPECT_EQ("ebf58670ee1512d597876fe632e92790bd70568374bdbe5a69c5d8ed107607ee", output.transaction_id());
+    EXPECT_EQ(169884, output.fee());
+    EXPECT_EQ("", output.error());
+}
+
+TEST(CardanoSigner, SignNoChange) {
```

Comment:
do this mean send all amounts? we have `use_max_amount` in Bitcoin.proto to handle sending max amount